apps/org/views.py

```diff
@@ -480,21 +483,61 @@ def form_valid(self, form):
         resource_request.save()
 
         # 2.) Create a ResourceGrant object for the ResourceRequest and the Member
-        ResourceGrant.objects.create(
+        ResourceGrant.objects.get_or_create(
             organization=resource_request.organization,
             member=resource_request.member,
             resource_class_path=resource_request.resource_class_path,
             resource_request=resource_request
         )
 
-        # 3.) Set the member's password
-        self.member.user.set_password(form.data['password1'])
-        self.member.user.save()
+        # 3.) Verify that the request.user has a UserSocialAuth object for VMI
+        request_user_social_auth = self.request.user.social_auth.filter(
+            provider=settings.SOCIAL_AUTH_NAME
+        ).first()
+        # If the request.user does not have a UserSocialAuth for VMI, then
+        # return the error to the user.
+        if not request_user_social_auth:
+            self.errors = {
+                'user': 'User has no association with {}'.format(settings.SOCIAL_AUTH_NAME)
+            }
+            return self.render_to_response(self.get_context_data())
 
-        # 4.) Redirect the user to the next step in the Member-creation process
-        return HttpResponseRedirect(
-            self.get_success_url(self.organization.slug, self.member.user.username)
-        )
+        # 4.) Verify that the Member has a UserSocialAuth object for VMI
+        member_social_auth = self.member.user.social_auth.filter(
+            provider=settings.SOCIAL_AUTH_NAME
+        ).first()
+        # If the Member does not have a UserSocialAuth for VMI, return an error to the user.
+        if not member_social_auth:
+            self.errors = {
+                'member': 'Member has no association with {}'.format(settings.SOCIAL_AUTH_NAME)
+            }
+            return self.render_to_response(self.get_context_data())
+
+        # 5.) Make a request to VMI to update the member's password in VMI
+        data = {'password': form.data['password1']}
+        url = '{}/api/v1/user/{}/'.format(settings.SOCIAL_AUTH_VMI_HOST, member_social_auth.uid)
+        headers = {'Authorization': "Bearer {}".format(request_user_social_auth.access_token)}
+        response = requests.put(url=url, data=data, headers=headers)
+
+        if response.status_code == 200:
+            # 6.) Set the member's password locally
+            self.member.user.set_password(form.data['password1'])
```

Comment:
I'd suggest leaving this out for now so that only one authorization path exists.

Reply:
Leave what out?  Seems necessary to me.

Reply:
I think the idea is that once the user sets their password in VMI, we don't need to set the same password in smh_app, since smh_app uses VMI to authenticate

Reply:
yes that is right.  There shouldn't be a local password in this case since smh_app a relying party.  Local users can still be used for administrative purposes.

Reply:
🆗 updated